cmd/snap-fde-keymgr/main.go

```diff
@@ -76,8 +80,24 @@ func (c *cmdAddRecoveryKey) Execute(args []string) error {
 	// 3. add the key
 	// 4. if adding failed with keyslot already in used and the file was
 	// present assume it's correct
-	if err := keymgrAddRecoveryKeyToLUKSDevice(recoveryKey, c.Device); err != nil {
-		return fmt.Errorf("cannot add recovery key to LUKS device: %v", err)
+	if len(c.Authorization) != len(c.Device) {
+		return fmt.Errorf("cannot add recovery key with mismatched devices and authorization")
```

Comment:
nitpick: I don't know how these keys look like, but this error message would prompt me to check something in the keys themselves, rather than just checking the number of devices and authorizations. Maybe "cannot add recovery key: mismatching number of devices and authorizations"?